### PR TITLE
Turn off Async suffix warning, and treat warnings as errors in CI build

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Let's get rid of those annoying warnings that clutter CI and our build logs.

### Changes

- Turn off the warning '"Async" suffix required for Task methods' in all projects
- Add some one time warning disable pragma's at sync/async boundaries.

### Testing
- Turn on treat warnings as errors in CI

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyTestGeneration/BlockBasedModification.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public class BlockBasedModification : ProgramModification {
       string? line;
       var stringReader = new StringReader(log);
       var newBlocksCovered = false;
-      while ((line = stringReader.ReadLine()) != null) {
+      while ((line = await stringReader.ReadLineAsync()) != null) {
```
Comment (cpitclaudel) [inline Source/DafnyTestGeneration/BlockBasedModification.cs:34]: Is this useful / necessary?

File: Source/DafnyTestGeneration/DafnyTestGeneration.csproj [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+    <DefineConstants>TRACE</DefineConstants>
+    <TargetFramework>net6.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ValidateExecutableReferencesMatchSelfContained>false</ValidateExecutableReferencesMatchSelfContained>
+  </PropertyGroup>
+  
+  <ItemGroup>
+    <ProjectReference Include="..\DafnyServer\DafnyServer.csproj" />
+    <ProjectReference Include="..\DafnyPipeline\DafnyPipeline.csproj" />
+  </ItemGroup>
```
Comment (cpitclaudel) [inline Source/DafnyTestGeneration/DafnyTestGeneration.csproj:17]: Why did you reindent this file?

**Review by cpitclaudel — APPROVED** [review]
Left one comment, but looks good to go

File: Source/DafnyTestGeneration/DafnyTestGeneration.csproj (hunk 2 above)
Comment (keyboardDrummer) [reply Source/DafnyTestGeneration/DafnyTestGeneration.csproj:17]: To be consistent with other .csproj files.

File: Source/DafnyTestGeneration/BlockBasedModification.cs (hunk 1 above)
Comment (keyboardDrummer) [reply Source/DafnyTestGeneration/BlockBasedModification.cs:34]: In general it improves performance. Rider presents it as a warning.

**Review by cpitclaudel — APPROVED** [review]

**Pull request merged**